test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/CharArrayWriterTest.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.io
+
+import java.io.{CharArrayWriter, StringWriter}
+import org.junit.Assert._
+import org.junit.Test
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class CharArrayWriterTest {
+  private val hw: Array[Char] = Array('H', 'e', 'l', 'l', 'o', 'W', 'o', 'r', 'l', 'd')
+
+  private def withClose[T <: AutoCloseable](closeable: T)(fn: T => Unit): Unit = {
+    fn(closeable)
+
+    if (closeable != null) closeable.close()
+  }
+
+  @Test def should_construct_with_correct_size_when_supplied(): Unit =
+    withClose(new CharArrayWriter(90)) { cw =>
+      assertEquals(0, cw.size)
+    }
+
+  @Test def should_construct_with_exception(): Unit = {
+    assertThrows(classOf[IllegalArgumentException], new CharArrayWriter(-1))
+  }
+
+  @Test def should_construct_with_correct_size_default(): Unit =
+    withClose(new CharArrayWriter) { cw =>
+      assertEquals(0, cw.size)
+    }
+
+  @Test def should_support_close(): Unit =
+    withClose(new CharArrayWriter) { cw =>
+      cw.close()
+      cw.close() // no-op
+      assertEquals(0, cw.size)
+    }
+
+  @Test def should_support_flush(): Unit =
+    withClose(new CharArrayWriter) { cw =>
+      cw.flush()
+    }
+
+  @Test def should_support_reset(): Unit =
+    withClose(new CharArrayWriter) { cw =>
+      cw.write("HelloWorld", 5, 5)
+      cw.reset()
+      cw.write("HelloWorld", 0, 5)
+
+      assertArrayEquals("Hello".toCharArray, cw.toCharArray)
+    }
+
+  @Test def should_support_size(): Unit =
+    withClose(new CharArrayWriter) { cw =>
+      assertEquals(0, cw.size)
+      cw.write(hw, 5, 5)
+      assertEquals(5, cw.size)
+    }
+
+  @Test def should_support_toCharArray(): Unit =
+    withClose(new CharArrayWriter) { cw =>
+      cw.write("HelloWorld", 0, 10)
+      assertArrayEquals("HelloWorld".toCharArray, cw.toCharArray)
+    }
+
+  @Test def should_support_toString(): Unit =
+    withClose(new CharArrayWriter) { cw =>
+      cw.write("HelloWorld", 5, 5)
+      assertEquals("World", cw.toString)
+    }
+
+  @Test def should_support_write_char_array(): Unit =
+    withClose(new CharArrayWriter) { cw =>
+      cw.write(hw, 5, 5)
+      assertEquals("World", cw.toString)
+      assertArrayEquals("World".toCharArray, cw.toCharArray)
+    }
+
+  @Test def write_should_throw_IndexOutOfBoundsException(): Unit =
+    withClose(new CharArrayWriter) { obj =>
+      assertThrows(
+        classOf[IndexOutOfBoundsException],
+        obj.write(Array[Char]('0'), 0, -1)
+      )
```

Comment:
It's not really well explained in our coding style document, but for long method calls that do not fit on one line, we use 4 spaces of indentation for continuation lines, and the `)` is on the same line as the last argument (except possibly if the arguments are a varargs):
```suggestion
      assertThrows(classOf[IndexOutOfBoundsException],
          obj.write(Array[Char]('0'), 0, -1))
```